hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -7665,7 +7670,7 @@ void addCachePool(CachePoolInfo req, boolean logRetryCache)
     checkOperation(OperationCategory.WRITE);
     String poolInfoStr = null;
     try {
-      checkSuperuserPrivilege();
+      checkSuperuserPrivilege(operationName);
```

Comment:
should we add the cache pool name as part of the context info as normal HDFS audit has that?

Reply:
That's a good suggestion @xiaoyuyao. We cannot add the cache pool name for "addCachePool()" method since the superuser check needs to happen before adding the cache pool and we get the name only after it's added. But, I will set the cache pool name in context for modify and remove methods.

Reply:
Can we get it from other calls after Line 7670?
String poolNameStr = "{poolName: " +
        (req == null ? null : req.getPoolName()) + "}";